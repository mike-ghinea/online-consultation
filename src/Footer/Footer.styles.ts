import styled from "styled-components";

const FooterWrapper = styled.footer`
  margin-top: auto;
  background-color: rgb(0, 63, 130);
  color: white;
  display: grid;
  line-height: 1.5;
  grid-template-columns: 1fr;
  @media (min-width: 480px) {
    grid-template-columns: 1fr 1fr;
  }
`;

const Section = styled.div`
  display: flex;
  flex-direction: column;
  padding: 2rem;
`;

const SectionHeader = styled.div`
  border-bottom: 1px solid #fff;
  font-size: 1.5rem;
`;

const SectionEntries = styled.div`
  display: flex;
  flex-direction: column;
  & > p {
    margin: 0;
  }
`;

const Link = styled.a`
  font-size: 1rem;
  color: white;
  text-decoration: none;
  &:hover,
  &:focus {
    text-decoration: underline;
  }
`;

export default { FooterWrapper, Section, SectionHeader, SectionEntries, Link };
