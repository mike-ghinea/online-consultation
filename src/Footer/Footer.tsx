import s from "./Footer.styles";

const Footer: React.FC = () => {
  return (
    <s.FooterWrapper>
      <s.Section>
        <s.SectionHeader>Useful links</s.SectionHeader>
        <s.SectionEntries>
          <s.Link href="#">Our treatments</s.Link>
          <s.Link href="#">Our team</s.Link>
          <s.Link href="#">Delivery Guarantee</s.Link>
          <s.Link href="#">Customer Testimonials</s.Link>
          <s.Link href="#">FAQ</s.Link>
          <s.Link href="#">Coming soon</s.Link>
        </s.SectionEntries>
      </s.Section>
      <s.Section>
        <s.SectionHeader>Contact Us</s.SectionHeader>
        <s.SectionEntries>
          <p>Mobile: 07000 000 000</p>
          <p>Landline: 0200 000 0000</p>
          <p>-----------------------</p>
          <p>info@example.com</p>
        </s.SectionEntries>
      </s.Section>
    </s.FooterWrapper>
  );
};

export default Footer;
