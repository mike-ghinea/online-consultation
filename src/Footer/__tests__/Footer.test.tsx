import Footer from "../Footer";
import { render } from "@testing-library/react";

describe("Footer", () => {
  it("should render correctly", () => {
    const { baseElement } = render(<Footer />);
    expect(baseElement).toMatchSnapshot();
  });
});
