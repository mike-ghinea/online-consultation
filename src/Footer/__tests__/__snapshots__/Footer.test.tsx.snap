// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<body>
  <div>
    <footer
      class="sc-blHHSb hcIBfi"
    >
      <div
        class="sc-gtLWhw bUoMZA"
      >
        <div
          class="sc-egkSDF fOCywZ"
        >
          Useful links
        </div>
        <div
          class="sc-fAUdSK jqQrVA"
        >
          <a
            class="sc-dntaoT iATqZd"
            href="#"
          >
            Our treatments
          </a>
          <a
            class="sc-dntaoT iATqZd"
            href="#"
          >
            Our team
          </a>
          <a
            class="sc-dntaoT iATqZd"
            href="#"
          >
            Delivery Guarantee
          </a>
          <a
            class="sc-dntaoT iATqZd"
            href="#"
          >
            Customer Testimonials
          </a>
          <a
            class="sc-dntaoT iATqZd"
            href="#"
          >
            FAQ
          </a>
          <a
            class="sc-dntaoT iATqZd"
            href="#"
          >
            Coming soon
          </a>
        </div>
      </div>
      <div
        class="sc-gtLWhw bUoMZA"
      >
        <div
          class="sc-egkSDF fOCywZ"
        >
          Contact Us
        </div>
        <div
          class="sc-fAUdSK jqQrVA"
        >
          <p>
            Mobile: 07000 000 000
          </p>
          <p>
            Landline: 0200 000 0000
          </p>
          <p>
            -----------------------
          </p>
          <p>
            info@example.com
          </p>
        </div>
      </div>
    </footer>
  </div>
</body>
`;
